vyper/parser/parser_utils.py

```diff
@@ -37,6 +36,17 @@
 getcontext().prec = 78  # MAX_UINT256 < 1e78
 
 
+def type_check_wrapper(fn):
+
+    def _wrapped(*args, **kwargs):
+        return_value = fn(*args, **kwargs)
+        if return_value is None:
+            raise TypeCheckFailure(f"{fn.__name__} did not return a value")
+        return return_value
+
+    return _wrapped
+
+
```

Comment:
I definitely don't like this, very magical

Reply:
If it's only temporary, it's okay I suppose

Reply:
All of it is temporary, I hope :)

_"All of this is a stop-gap on the way to a full refactor of the `parser/` module - we just need to make sure that if the type check layer misses something, the same exception isn't raised in `parser/` so that we can address the error."_